src/js/07-theme-toggle.js

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-undef */
+var toggle = document.getElementById('theme-toggle')
+
+function setIcon (theme) {
+  if (theme === 'dark') {
+    toggle.innerHTML = '<span class="material-icons">light_mode</span>'
+  } else {
+    toggle.innerHTML = '<span class="material-icons">dark_mode</span>'
+  }
+}
+
+var storedTheme =
+  localStorage.getItem('theme') ||
+  (window.matchMedia('(prefers-color-scheme: dark)').matches
```

Comment:
One problem I'm still looking into is the "flicker" that happens when your system/browser prefers dark mode. Light mode is the :root css default so the page flashes white initially before the JS code sees that it should be dark and switches